### ISSUE TITLE
[ENH] turning `Pandas` into a soft dependency

### ISSUE THREAD

**Issue by patrickzib** [sktime/sktime#3873]
Following the same logic as in https://github.com/sktime/sktime/issues/3567 I hereby propose:

I think we should turn `Pandas` into a soft dependency, since it is not core to the framework (it is possible to run any estimator with `numpy` input), but used only in specific estimators / situations. 

---

key arguments for:

* the only key change is giving users the choice `Pandas` yes/no
* for users with `Pandas` installed, the `sktime` experience remains precisely the same
  * users without `Pandas` will be able to use `sktime`, currently they cannot
  * users without `Pandas` will still be able to find `Pandas` based estimators using `all_estimators`, similar to other soft dependencies
* it should be easier to upgrade `sktime` to newer python or package versions, of core dependencies, as `Pandas` is slow to update compatibility
* Pandas is constantly broken
* architectural: reduces coupling, increases layer separation; minimize soft dependencies, see context below

key arguments against:

I don't really see any except the work required

---

Update: more context on architectural and usage benefits.

the general desideratum is "minimize dependencies". This is an explicit architectural north star of scikit-learn as well as sktime.

See e.g., this paper on sklearn design,
https://mblondel.org/publications/lbuitinck-ecmlpkdd2013.pdf
search for "dependencies" in the text.

In the past 6 months, we have made the interfaces of estimators much clearer, including that we can manage python versions and package dependencies on the estimator level, rather than at the sktime package level

This has prompted multiple lines of refactoring around "minimize dependencies":

* migrate sktime-dl into sktime because we can isolate tensorflow to estimators
* isolate `statsmodels`
* newly added annotation module soft dependencies are fully isolated

`statsmodels` isolation and soft dep is very similar, see this PR: https://github.com/sktime/sktime/pull/3821

---

FAQ

> does this remove `Pandas` support from `sktime`?

No, `Pandas` will still be supported and used in estimators.
It will only change to be optional, for people who do not want (or need) to use estimators that depend on `Pandas`.

> will this impact performance of `sktime`?

No. For users who have `Pandas` installed, the experience is precisely the same. The logic of `Pandas` based estimators stays the same.

No change in estimator accuracy or speed. (evidence: look at the tests in https://github.com/sktime/sktime/pull/3843 which test for both)

The only change is that users get a choice whether they want `Pandas` or not.

**Comment by fkiraly** [comment]
I think we can close this in favour of https://github.com/sktime/sktime/issues/3876

**Comment by patrickzib** [comment]
Objection your honour.

**Comment by patrickzib** [comment]
Let`s move it to stage 2 then.

**Comment by fkiraly** [comment]
probably trolling/disruption. Will be discussed internally.
See https://github.com/sktime/sktime/issues/3882#issuecomment-1336287948

Formally, you are objecting to your own issue, you cannot trigger stage 2 resolution that way.
